docs/release-notes/identify-task.rst

```diff
@@ -0,0 +1,7 @@
+:orphan:
+
+**Bug Fixes**
+
+-  API: Fix a bug where a database query would add extra quotes around the task type.
+      For tensorboard task types, this bug would always error and not allow authorized users to view
```

Comment:
this wording is a little confusing

Also isn't the bug the opposite of this? That users could view tensorboards that they shouldn't have been able to?

Reply:
tagged @tara-det-ai in this PR to help with the wording of this, but tbh I'm not quite sure which way the error would go, given that CanGetTensorboard would check the permission `PermissionType_PERMISSION_TYPE_VIEW_WORKSPACE`, whereas CanGetNSC would check permission `PermissionType_PERMISSION_TYPE_VIEW_NSC`. I don't know in tensorboards would have the default permission to view NSCs.

Reply:
My understanding of the bug is that pre this change we always run the ```CanGetNSC``` check


```
	if spec.TaskType == model.TaskTypeTensorboard { // the bug is spec
		err = command.AuthZProvider.Get().CanGetTensorboard(
			ctx, *user, spec.WorkspaceID, spec.ExperimentIDs, spec.TrialIDs)
	} else {
		err = command.AuthZProvider.Get().CanGetNSC(
			ctx, *user, spec.WorkspaceID)
	}
```

```CanGetNSC``` is basically if you have viewer on a workspace

Post this change we will now call ```CanGetTensorboard(..., spec.TrialIDs)``` which checks the workspace IDs of every trial specified for the tensorboard

https://github.com/determined-ai/determined-ee/blob/1628010f54abcb80d0bac55347437e8a7ef4f66c/master/internal/command/authz_rbac.go#L208-L217

So if user A creates a tensorboard with experiments from workspace 1 and 2, and user B only can view workspace 1 then after this change they will correctly no longer be able to view the tensorboard? Does that line up to what you think the bug was?

Reply:
As the release note is written I almost get the impression that tensorboards are broken for all of RBAC.

Reply:
Yeah that makes sense looking at the code, updated the note